docs/posts/flink-quickstart/index.qmd

```diff
@@ -0,0 +1,262 @@
+```
+
+## Using `mamba`
+
+```bash
+mamba install -c conda-forge ibis-flink kafka-python
+```
+
+## Using `pixi`
+
+```bash
+pixi add ibis-flink kafka-python
+```
+
+:::
+
+## Spinning up the services using Docker Compose
+
+The [claypotai/ibis-flink-example GitHub repository](https://github.com/claypotai/ibis-flink-example) includes the relevant Docker Compose configuration for this tutorial. Clone the repository, and run `docker compose up` from the cloned directory to create Kafka topics, generate sample data, and launch a Flink cluster.
+
+```bash
+git clone https://github.com/claypotai/ibis-flink-example.git
+cd ibis-flink-example
+docker compose up
+```
+
+::: {.callout-tip}
+If you don't intend to try [remote execution](#remote-execution), you can start only the Kafka-related services with `docker compose up kafka init-kafka data-generator`.
+:::
+
+After a few seconds, you should see messages indicating your Kafka environment is ready:
+
+```bash
+ibis-flink-example-init-kafka-1      | Successfully created the following topics:
+ibis-flink-example-init-kafka-1      | payment_msg
+ibis-flink-example-init-kafka-1      | sink
+ibis-flink-example-init-kafka-1 exited with code 0
+ibis-flink-example-data-generator-1  | Connected to Kafka
+ibis-flink-example-data-generator-1  | Producing 20000 records to Kafka topic payment_msg
+```
+
+The `payment_msg` Kafka topic contains messages in the following format:
+
+```json
+{
+    "createTime": "2023-09-20 22:19:02.224",
+    "orderId": 1695248388,
+    "payAmount": 88694.71922270155,
+    "payPlatform": 0,
+    "provinceId": 6
+}
+```
+
+In a separate terminal, we can explore what these messages look like:
+
+```{python}
+from kafka import KafkaConsumer
+
+consumer = KafkaConsumer("payment_msg")
+for _, msg in zip(range(3), consumer):
+    print(msg)
+```
+
+## Running the tutorial
+
+This tutorial uses the Flink backend for Ibis to process the aforementioned payment messages. You can choose to either [run it locally](#local-execution) or [submit a job to an already-running Flink cluster](#remote-execution).
+
+### Local execution
+
+The simpler option is to run the example using the Flink mini cluster.
+
+#### Create a table environment
+
+The [table environment](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/python/table/table_environment/) serves as the main entry point for interacting with the Flink runtime. The `flink` backend does not create `TableEnvironment` objects; you must create a `TableEnvironment` and pass that to [`ibis.flink.connect`](../../backends/flink.qmd#ibis.flink.connect).
+
+```{python}
+import ibis
+from pyflink.table import EnvironmentSettings, TableEnvironment
+
+env_settings = EnvironmentSettings.in_streaming_mode()
+table_env = TableEnvironment.create(env_settings)
+# write all the data to one file
+table_env.get_config().set("parallelism.default", "1")
+
+connection = ibis.flink.connect(table_env)
```

Comment:
We tend to call this `con` in I think every other bit of docs, let's try to stay consistent here.